src/main/java/com/google/sps/server/ScheduleCreationHandler.java

```diff
@@ -14,64 +14,58 @@
 
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Arrays;
 
 import com.google.sps.data.*;
 import com.google.sps.Scheduler;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
 
 /**
  * Create and rank the schedules     
  */
-public class ScheduleCreationHandler {
+@WebServlet("/handleUserInput")
+public class ScheduleCreationHandler extends HttpServlet {
   
-  private static final int DURATION_90_MINUTES = 90;
-  private static final int DURATION_3_HOUR = 180;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    JsonObject wholeJSON;
+    try {
+      wholeJSON = JsonParser.parseReader(request.getReader()).getAsJsonObject();
+    } catch (Exception e) {
+      System.out.println("Error reading JSON: " + e.getMessage());
+      return; //TODO: Handle this properly
+    }
 
-  public static List<Schedule> createSchedules() {
-    // Currently using hard-coded courses and preferences 
-    List<Course> courses = createCoursesPriority();
-    List<TimeRange> userNoClassTimes = TesterSchedule.monWedFri(7, 00, DURATION_3_HOUR);
-    List<Course> courseList = createCoursesPriority();
-    String preferredSubject = "Computer Science";
+    Gson gson = new Gson();
+    System.out.println(wholeJSON.toString());
 
-    // Permute the courses to create the schedules 
-    Scheduler scheduler = new Scheduler();
-    List<Schedule> schedules = scheduler.generateSchedules(courses, new Invariants(10,54));
-
-    PrioritizeCoursesCriteria coursePriority = new PrioritizeCoursesCriteria(courseList);
-    RestrictTimesCriteria restrictCriteria = new RestrictTimesCriteria(userNoClassTimes);
-    SubjectCoursesCriteria subjectCourses = new SubjectCoursesCriteria(preferredSubject);
-    Preferences preferenceList = new Preferences(Arrays.asList(coursePriority, restrictCriteria, subjectCourses));
-
-    // Rank the schedules
-    preferenceList.sortSchedules(schedules);
-    return schedules;
-  }
-
-   /**
-   * Helper function to instantiate hard-coded courses, and their preferred order 
-   *
-   * @return a List of the ordered Course objects      
-   */
-  private static List<Course> createCoursesPriority() {
-    Section section1 = new Section("Professor A", TesterSchedule.monWedFri(9, 00, DURATION_90_MINUTES));
-    Course course1 = new Course("Operating System", "15410", 
-        "Computer Science", 15, true, Arrays.asList(section1));
+    Type courseListType = new TypeToken<List<Course>>(){}.getType();
+    List<Course> courses = gson.fromJson(wholeJSON.getAsJsonArray("courses"), courseListType);
```

Comment:
It would be cleaner if you had a single `GenerateScheduleRequest` type that contained all of the fields you care about (courses, basicInfo, credits...). You could then do:

```
GenerateScheduleRequest request = gson.fromJson(wholeJSON, GenerateScheduleRequest.class);
List<Course> courses = request.getCourses();
Invariants invariants = request.getCredits();
...
```

Something like that. One single structure to deserialize and then everything is normal type-checked Java from there on out.

It's much easier to work with a single class definition that you then serialize to and from.